### ISSUE TITLE
[BUG] Missing packages for 'StatusNotifierWatcher'?

### ISSUE THREAD

**Issue by sbradnick** [elkowar/eww#1170]
### Checklist before submitting an issue

- [X] I have searched through the existing [closed and open issues](https://github.com/elkowar/eww/issues?q=is%3Aissue) for eww and made sure this is not a duplicate
- [X] I have specifically verified that this bug is not a common [user error](https://github.com/elkowar/eww/issues?q=is%3Aissue+label%3Ano-actual-bug+is%3Aclosed)
- [X] I am providing as much relevant information as I am able to in this bug report (Minimal config to reproduce the issue for example, if applicable)

### Description of the bug

I have a very basic tray implementation which was working fine ahead of the latest set of commits being merged; after building (`cargo build --release --no-default-features --features x11`) when it starts it's empty and prints the following:
```
eww::widgets::systray  > could not initialise dbus connection for tray: org.freedesktop.DBus.Error.UnknownObject: Unknown object '/org/freedesktop/StatusNotifierWatcher'
```

### Reproducing the issue

eww daemon ; eww open tray

### Expected behaviour

A tray window w/ items in it as it was working before.

### Additional context

I'm doing various `git checkout <hash>` runs to see if I can find which commit broke it. Not sure I have the proper skillset to fix anything :P ...

**Comment by sbradnick** [comment]
OS: openSUSE Tumbleweed

Here's all I have for the tray:
```
(defwidget mysystray []
  (systray)
)
(defwindow tray
  :geometry
  (geometry
    :anchor "center right"
    :x "-170px"
    :y "45%"
  )
  :orientation "v"
  :stacking "bg"
  :windowtype "normal"
  (box
    :class "bx"
    :orientation "v"
    :space-evenly false
    (mysystray)
  )
)
```

**Comment by w-lfchen** [comment]
just out of curiosity: does compiling with default features (simply `cargo build --release`) work?

if that doesn't fix it, i'd assume the issue is some package missing or some service not running, but i sadly know nothing about your distro :/

**Comment by sbradnick** [comment]
`cargo build --release` works after installing what provides 'pkgconfig(gtk-layer-shell-0)', errored out initially; but the resulting binary still complains about the tray.

Is there "something" that generally provides this `Unknown object '/org/freedesktop/StatusNotifierWatcher'`? Some dbus command that needs run or maybe there's some config file I'm missing that it's now looking for?

openSUSE Tumbleweed is a rolling release that uses RPMs. Built from submissions here: https://build.opensuse.org/project/show/openSUSE:Factory ; not that I expect you to dig into TW for anything related to this, was just hoping to find out what's changed between eda9d1e -> 71ba502 that would make it emit this error.

**Comment by w-lfchen** [comment]
71ba5024421dd475e800c55f3efce1dd887cdc56 updated the crate managing dbus with a major version bump which had a lot of breaking changes. i did my best to follow the documentation on updating when working on that, but i can't rule out that i missed something.

however, i'm really confused about the fact that i am not getting any errors on my system, i'm running nixos from a fairly recent version

**Comment by w-lfchen** [comment]
i broke this, so i will have to fix it, see #1180 for more information.
if i can't find the cause, i'll revert the update which should work but is obviously not ideal

**Comment by sbradnick** [comment]
Back to normal on my end too 👍 - glad it wasn't an OS/dbus-specific problem I needed to struggle with - appreciate your time in finding the resolution.
